app/src/main/java/com/example/yard/fragments/HomeFragment.java

```diff
@@ -0,0 +1,141 @@
+
+import com.example.yard.R;
+import com.example.yard.adapter.PostCreation;
+import com.example.yard.adapter.PostsAdapter;
+import com.example.yard.helperclass.PostQueryHelper;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class HomeFragment extends Fragment {
+  public static final String TAG = "HomeFragment";
+  Handler handler;
+  ProgressBar fetchMoreBar;
+  NestedScrollView nestedScrollView;
+  int value;
+  boolean hasFetchedData = false;
+  private PostsAdapter adapter;
+  private List<PostCreation> allPosts;
+  private PostQueryHelper queryPosts;
+  private ProgressBar loadingBar;
+  private HandlerThread backgroundThread;
+  private Context context;
+
+  public HomeFragment() {}
+
+  @Override
+  public void onAttach(@NonNull Context context) {
+    super.onAttach(context);
+    this.context = context;
+  }
+
+  @Nullable
+  public View onCreateView(
+      LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+    return inflater.inflate(R.layout.fragment_home, container, false);
+  }
+
+  @Override
+  public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+    super.onViewCreated(view, savedInstanceState);
+
+    backgroundThread = new HandlerThread("backgroundThread");
+    backgroundThread.start();
+    handler = new Handler(backgroundThread.getLooper());
+
+    RecyclerView rvPosts = getView().findViewById(R.id.rvPosts);
+    loadingBar = view.findViewById(R.id.progressBarId);
+    fetchMoreBar = view.findViewById(R.id.fetchMoreBar);
+    nestedScrollView = view.findViewById(R.id.scrollBar);
+    allPosts = new ArrayList<>();
+    adapter = new PostsAdapter(getContext(), allPosts);
+    rvPosts.setAdapter(adapter);
+    rvPosts.setLayoutManager(new LinearLayoutManager(getContext()));
+    queryPosts = new PostQueryHelper();
+    startThread();
+    nestedScrollView.setOnScrollChangeListener(
+        new NestedScrollView.OnScrollChangeListener() {
+          @Override
+          public void onScrollChange(
+              NestedScrollView v, int scrollX, int scrollY, int oldScrollX, int oldScrollY) {
+            if (scrollY == v.getChildAt(0).getMeasuredHeight() - v.getMeasuredHeight()) {
+              fetchMoreBar.setVisibility(View.VISIBLE);
+              startThread();
+            }
+          }
+        });
+  }
+
+  public void startThread() {
+    startActivityBackground startActivity = new startActivityBackground();
+    new Thread(startActivity).start();
+  }
+
+  public Boolean showProgressBar() {
+    return true;
+  }
+
+  public class startActivityBackground implements Runnable {
+    @Override
+    public void run() {
+      handler.post(
+          new Runnable() {
+            @Override
+            public void run() {
+              if (showProgressBar()) {
+                queryPosts.postQueryHelper(false, adapter, allPosts, new OnQueryDone());
+              }
+              hasFetchedData = true;
```

Comment:
Query a post doesn't gaurantee posts has been successfully fetched. Post the changing of the boolean flag to after post fetching succeeds.